payment-request/constructor_convert_method_data.https.html

```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE html> <meta charset="utf-8" />
+<title>Validates PaymentMethodData's data member during construction</title>
+<link
+  rel="help"
+  href="https://w3c.github.io/browser-payment-api/#constructor"
+/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  const details = {
+    total: {
+      label: "Total",
+      amount: {
+        currency: "USD",
+        value: "0.00",
+      },
+    },
+  };
+
+  test(() => {
+    new PaymentRequest([{ supportedMethods: "basic-card" }], details);
+    new PaymentRequest(
+      [{ supportedMethods: "https://apple.com/apple-pay" }],
+      details
+    );
+  }, "Smoke test.");
+
+  const knownPMIs = ["basic-card", "https://apple.com/apple-pay"];
+  const unknownPMIs = ["fake-pmi", "https://does-not.exist"];
+  promise_test(async t => {
+    for (const supportedMethods of [...knownPMIs, ...unknownPMIs]) {
+      const method = { supportedMethods };
+      const request = new PaymentRequest([method], details);
+      assert_throws(
+        new TypeError(),
+        () => {
+          const badMethod = {
+            ...method,
```

Comment:
I apologize I oversaw this, spread object properties won't work in Edge. Maybe Object assign?

Reply:
No probs. Yeah, will do Object.assign.